src/graph/01_bfs.md

```diff
@@ -37,14 +37,14 @@ while (!q.empty()) {
 
 We can notice that the difference between the distances between the source `s` and two other vertices in the queue differs by at most one.
 Especially, we know that $d[v] \le d[u] \le d[v] + 1$ for each $u \in Q$.
-The reason for this is, that we only add vertices with equal distance or with distance plus one to the queue each iteration.
+The reason for this is, that we only add vertices with equal distance or with distance plus one to the queue for each iteration.
 Assuming there exists a $u$ in the queue with $d[u] - d[v] > 1$, then $u$ must have been insert in the queue via a different vertex $t$ with $d[t] \ge d[u] - 1 > d[v]$.
 However this is impossible, since Dijkstra's algorithm iterates over the vertices in increasing order.
 
 This means, that the order of the queue looks like this:
 $$Q = \underbrace{v}\_{d[v]}, \dots, \underbrace{u}\_{d[v]}, \underbrace{m}\_{d[v]+1} \dots \underbrace{n}\_{d[v]+1}$$
 
-This structure is so simple, that we don't need an actually priority queue, i.e. a balanced binary tree, is an overkill.
+This structure is so simple, that we don't need an actual priority queue, i.e. a balanced binary tree which is overkill.
```

Comment:
I agree, ", is an overkill" doesn't fit at all in the sentence.
But your change is also not good. I would suggest something like "i.e. using a balanced binary tree would be an overkill".